services/upgrader/upgrader_linux.go

```diff
@@ -108,21 +109,51 @@ func verifySignature(filename string, sigfilename string, publicKey []byte) erro
 		mlog.Debug("Unable to open the Mattermost .tar file to verify the file signature", mlog.Err(err))
 		return NewInvalidSignature()
 	}
+	defer mattermost_tar.Close()
 
 	signature, err := os.Open(sigfilename)
 	if err != nil {
-		mlog.Debug("Unable to open the Mattermost .sig file verify the file signature", mlog.Err(err))
+		mlog.Debug("Unable to open the Mattermost .sig file to verify the file signature", mlog.Err(err))
 		return NewInvalidSignature()
 	}
+	defer signature.Close()
 
-	_, err = openpgp.CheckDetachedSignature(keyring, mattermost_tar, signature)
+	decodedSignature, err := decodeArmoredSignature(signature)
+	if err != nil {
+		mlog.Debug("Unable to decode the Mattermost .sig file to verify the file signature", mlog.Err(err))
+		return NewInvalidSignature()
+	}
+
+	_, err = openpgp.CheckDetachedSignature(keyring, mattermost_tar, decodedSignature)
```

Comment:
Do we need to execute `CheckDetachedSignature` when `decodeArmoredSignature` is successful ?

If error is nil for `decodeArmoredSignature`, `CheckDetachedSignature` will generate error.

Reply:
Yes, `decodeArmoredSignature` only sets up a decoder for the armored block, it doesn't validate the signature, it doesn't even validate that the complete signature can be decoded.